test/pre-post-extension.js

```diff
@@ -0,0 +1,56 @@
+import chai, {expect} from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import Stipulate from '../src';
+
+chai.use(chaiAsPromised);
+
+class ExtStip extends Stipulate {
```

Comment:
I think I should respond more specifically to a couple things you mentioned:
>You're testing the EMCA 6 specification for class extension, which shouldn't be tested.

It's not being tested, it is being used during setup of the tests. If you're saying the descriptions of the tests and the name of the file are confusing, I now understand why and agree.

>Extension is clearly defined and an expected behavior.

Definitely. I whole-heartedly agree. Which is why [your suggestion](https://github.com/yola/stipulate/pull/2#discussion_r59650144) to eschew its use (i.e. class inheritance) in favor of `_beforeRequest` which looks for `this.before` (or whatever else it ends up being named) doesn't make sense to me. It screams of over-engineering to me when there's a perfectly simple, safe, intelligible, native mechanism to use.